pyresample/future/spherical/extent.py

```diff
@@ -0,0 +1,300 @@
+    if extent[2] > extent[3]:
+        raise ValueError('extent[2] (lat_min) must be smaller than extent[3] (lat_max).')
+
+
+def _check_extent_values(extent, use_radians=False):
+    """Check extent value validity."""
+    lat_valid_range = [-90.0, 90.0]
+    lon_valid_range = [-180.0, 180.0]
+    extent = np.array(extent)
+    if use_radians:
+        lat_valid_range = np.deg2rad(lat_valid_range)
+        lon_valid_range = np.deg2rad(lon_valid_range)
+    if np.any(np.logical_or(extent[0:2] < lon_valid_range[0], extent[0:2] > lon_valid_range[1])):
+        raise ValueError(f'extent longitude values must be within {lon_valid_range}.')
+    if np.any(np.logical_or(extent[2:] < lat_valid_range[0], extent[2:] > lat_valid_range[1])):
+        raise ValueError(f'extent latitude values must be within {lat_valid_range}.')
+
+
+def _check_is_not_point_extent(extent):
+    """Check the extent does not represent a point."""
+    if extent[0] == extent[1] and extent[2] == extent[3]:
+        raise ValueError("An extent can not be defined by a point.")
+
+
+def _check_is_not_line_extent(extent):
+    """Check the extent does not represent a line."""
+    if extent[0] == extent[1] or extent[2] == extent[3]:
+        raise ValueError("An extent can not be defined by a line.")
+
+
+def _check_valid_extent(extent, use_radians=False):
+    """Check lat/lon extent validity."""
+    # Check extent length
+    if len(extent) != 4:
+        raise ValueError("'extent' must have length 4: [lon_min, lon_max, lat_min, lat_max].")
+    # Convert to np.array
+    extent = np.array(extent)
+    # Check order and values
+    _check_extent_order(extent)
+    _check_extent_values(extent, use_radians=use_radians)
+    # Check is not point or line
+    _check_is_not_point_extent(extent)
+    _check_is_not_line_extent(extent)
+    # Ensure is a list
+    extent = extent.tolist()
+    return extent
+
+
+def _check_non_overlapping_extents(polygons):
+    """Check that the extents polygons are non-overlapping.
+
+    Note: Touching extents are considered valids.
+    """
+    for poly in polygons:
+        # Intersects includes within/contain/equals/touches !
+        n_intersects = np.sum([p.intersects(poly) and not p.touches(poly) for p in polygons])
+        if n_intersects >= 2:
+            raise ValueError("The extents composing SExtent can not be duplicates or overlapping.")
+
+
+def _is_global_extent(polygons):
+    """Check if a list of extents composes a global extent."""
+    from shapely.geometry import Polygon
+    from shapely.ops import unary_union
+
+    # Try union the polygons
+    unioned_polygon = unary_union(polygons)
+    # If still a MultiPolygon, not a global extent
+    if not isinstance(unioned_polygon, Polygon):
+        return False
+    # Define global polygon
+    global_polygon = Polygon.from_bounds(-180, -90, 180, 90)
+    # Assess if global
+    is_global = global_polygon.equals(unioned_polygon)
+    return is_global
+
+
+class SExtent(object):
```

Comment:
I'm not sure if the other maintainers disagree, but I'm not a fan of the object subclass anymore.

```suggestion
class SExtent:
```